### PR TITLE
Tag autogenerated users using a settings.

### PR DESCRIPTION
**Pull request by mythmon**
This uses the user settings as an arbitrary key value store to record that a user was auto generated. Most of these changes are to get the test to work nicely.

r?

### CODE REVIEW

File: kitsune/users/tests/test_api.py [hunk 1]
```diff
@@ -101,6 +101,9 @@ def test_no_duplicate_emails(self):
     def test_users_without_emails_are_active(self):
         del self.data['email']
         serializer = api.ProfileSerializer(data=self.data)
+        import q
```
Comment (rehandalal) [inline kitsune/users/tests/test_api.py]: d'oh

Comment (rlr) [inline kitsune/users/tests/test_api.py]: GAH!

**Comment by mythmon** [comment]
Fixed ^

**Comment by rehandalal** [comment]
r+

**Pull request merged**